src/models/event-timeline-set.js

```diff
@@ -187,7 +187,7 @@ EventTimelineSet.prototype.resetLiveTimeline = function(backPaginationToken, flu
     newTimeline.setPaginationToken(backPaginationToken, EventTimeline.BACKWARDS);
 
     this._liveTimeline = newTimeline;
-    this.emit("Room.timelineReset", this.room, this);
+    this.emit("Room.timelineReset", this.room, this, flush);
```

Comment:
also, should be `flush || !this._timelineSupport`. Or better yet, factor out the condition so that they don't get out of sync.